cloud/amazon/ec2.py

```diff
@@ -1207,17 +1207,15 @@ def main():
 
     ec2 = ec2_connect(module)
 
-    ec2_url, aws_access_key, aws_secret_key, region = get_ec2_creds(module)
+    region, ec2_url, boto_params = get_aws_connection_info(module)
 
     if region:
         try:
-            vpc = boto.vpc.connect_to_region(
-                region,
-                aws_access_key_id=aws_access_key,
-                aws_secret_access_key=aws_secret_key
-            )
+            vpc = boto.vpc.connect_to_region(region, **boto_params)
```

Comment:
`vpc = connect_to_aws(boto.vpc, region, **boto_params)`

fixes a couple of other minor issues (particularly arround security tokens with profiles)